### PR TITLE
Update TaskResponse to have TaskNetworkConfig

### PR DESCRIPTION
**Pull request by xxx0624**
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/aws/amazon-ecs-agent/blob/master/CONTRIBUTING.md

Please provide the following information:
-->

### Summary
<!-- What does this pull request do? -->
This PR is to add the `TaskNetworkConfig` to the `TaskResponse` so that [GetTaskMetadata in AgentState](https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/tmds/handlers/v4/state/state.go#L104C2-L104C67) can help Faults handlers to get necessary task network configurations ie the task network mode, primary task network namespace (in awsvpc mode) and the primary task ENI (in awsvpc mode).

### Implementation details
<!-- How are the changes implemented? -->
1. Add the shared `TaskNetworkConfig` to the https://github.com/aws/amazon-ecs-agent/blob/master/ecs-agent/tmds/handlers/v4/state/response.go#L32

### Testing
<!-- How was this tested? -->

1. Add a unit test to make sure this new field won't be exposed to regular TMDS responses.
2. Manual tested that this new field won't show up in regular task metadata response.
  - Launch a task and check the task metadata response
 ```
sh-5.2# curl http://169.254.170.2/v4/metadata | jq .                                                                                                                                                                                                                                                                                                                                                                   
{                                                                                                                                                                                                                                                     
  "Cluster": "arn:aws:ecs:us-west-2:371230630865:cluster/tobedeletedsoon",                                                                                                                                                                            
  "TaskARN": "arn:aws:ecs:us-west-2:371230630865:task/tobedeletedsoon/a22f36c6d8b8442b80fb98e6add00afd",                                                                                                                                              
  "Family": "shim-logger-test",                                                                                                                                                                                                                       
  "Revision": "6",                                                                                                                                                                                                                                    
  "DesiredStatus": "RUNNING",                                                                                                                                                                                                                         
  "KnownStatus": "RUNNING",                                                                                                                                                                                                                           
  "Limits": {                                                                                                                                                                                                                                         
    "CPU": 0.5,                                                                                                                                                                                                                                       
    "Memory": 1024                                                                                                                                                                                                                                    
  },
  "PullStartedAt": "2024-08-08T00:57:43.097600256Z",
  "PullStoppedAt": "2024-08-08T00:57:47.32102539Z",
  "AvailabilityZone": "us-west-2a",
  "LaunchType": "FARGATE",
  "Containers": [
    {
      "DockerId": "a22f36c6d8b8442b80fb98e6add00afd-3147973833",
      "Name": "testing-1",
      "DockerName": "testing-1",
      "Image": "public.ecr.aws/amazonlinux/amazonlinux:2023",
      "ImageID": "sha256:03c807fdc5d40ab06b21c2e6d158c63c37d5149a127b7fdfa9e3c1150352ff20",
      "Labels": {
        "com.amazonaws.ecs.cluster": "arn:aws:ecs:us-west-2:371230630865:cluster/tobedeletedsoon",
        "com.amazonaws.ecs.container-name": "testing-1",
        "com.amazonaws.ecs.task-arn": "arn:aws:ecs:us-west-2:371230630865:task/tobedeletedsoon/a22f36c6d8b8442b80fb98e6add00afd",
        "com.amazonaws.ecs.task-definition-family": "shim-logger-test",
        "com.amazonaws.ecs.task-definition-version": "6"
      },
      "DesiredStatus": "RUNNING",
      "KnownStatus": "RUNNING",
      "Limits": {
        "CPU": 2
      },
      "CreatedAt": "2024-08-08T00:57:59.356321847Z",
      "StartedAt": "2024-08-08T00:57:59.356321847Z",
      "Type": "NORMAL",
      "LogDriver": "awslogs",
      "LogOptions": {
        "awslogs-create-group": "true",
        "awslogs-group": "xing-testing-group",
        "awslogs-region": "us-west-2",
        "awslogs-stream": "xing-testing-prefix/testing-1/a22f36c6d8b8442b80fb98e6add00afd",
        "mode": "non-blocking"
      },
      "ContainerARN": "arn:aws:ecs:us-west-2:371230630865:container/tobedeletedsoon/a22f36c6d8b8442b80fb98e6add00afd/7f36cd07-6879-42e1-84fd-86e5f821b8cf",
      "Networks": [
        {
          "NetworkMode": "awsvpc",
          "IPv4Addresses": [
            "10.194.20.187"
          ],
          "AttachmentIndex": 0,
          "MACAddress": "06:89:62:09:64:63",
          "IPv4SubnetCIDRBlock": "10.194.20.0/24",
          "PrivateDNSName": "ip-10-194-20-187.us-west-2.compute.internal",
          "SubnetGatewayIpv4Address": "10.194.20.1/24"
        }
      ],
      "Snapshotter": "overlayfs"
    }
  ],
  "ClockDrift": {
    "ClockErrorBound": 0.46121499999999993,
    "ReferenceTimestamp": "2024-08-08T00:59:51Z",
    "ClockSynchronizationStatus": "SYNCHRONIZED"
  },
  "EphemeralStorageMetrics": {
    "Utilized": 268,
    "Reserved": 20496
  }
}
sh-5.2#
 ```

<!--
Note for external contributors:
`make test` and `make run-integ-tests` can run in a Linux development
environment like your laptop.  `go test -timeout=30s ./agent/...` and
`.\scripts\run-integ.tests.ps1` can run in a Windows development environment
like your laptop.  Please ensure unit and integration tests pass (on at least
one platform) before opening the pull request.
Once you open the pull request, there will be 14 automatic test checks on the bottom
of the pull request, please make sure they all pass before you merge it. You can
use `bot/test` label to rerun the automatic tests multiple times.
-->

New tests cover the changes: <!-- yes|no -->
Yes

### Description for the changelog
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog.
You can see our changelog entry style here:
https://github.com/aws/amazon-ecs-agent/commit/c9aefebc2b3007f09468f651f6308136bd7b384f
-->
Feature - add an internal field in TaskResponse object.

**Does this PR include breaking model changes? If so, Have you added transformation functions?**
<!-- If yes, next release should have a upgraded minor version -->  
No

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/tmds/handlers/v4/handlers_test.go [hunk 1]
```diff
@@ -24,21 +24,23 @@ import (
 	"testing"
 	"time"
 
+	"github.com/aws/aws-sdk-go/aws"
```
Comment (mye956) [inline ecs-agent/tmds/handlers/v4/handlers_test.go:27]: nit(non-blocking): ordering of imports is a bit inconsistent with our existing format. (1. Built-in packages, 2. Local packages, 3. External packages)

File: ecs-agent/tmds/handlers/v4/state/response.go [hunk 2]
```diff
@@ -31,12 +32,13 @@ const (
...
-	ClockDrift              *ClockDrift              `json:"ClockDrift,omitempty"`
-	EphemeralStorageMetrics *EphemeralStorageMetrics `json:"EphemeralStorageMetrics,omitempty"`
-	CredentialsID           string                   `json:"-"`
+	Containers              []ContainerResponse                  `json:"Containers,omitempty"`
+	VPCID                   string                               `json:"VPCID,omitempty"`
+	ServiceName             string                               `json:"ServiceName,omitempty"`
+	ClockDrift              *ClockDrift                          `json:"ClockDrift,omitempty"`
+	EphemeralStorageMetrics *EphemeralStorageMetrics             `json:"EphemeralStorageMetrics,omitempty"`
+	CredentialsID           string                               `json:"-"`
+	TaskNetworkConfig       *tasknetworkconfig.TaskNetworkConfig `json:"-"`
```
Comment (mye956) [inline ecs-agent/tmds/handlers/v4/state/response.go:41]: Q: Should we also have a boolean in the task response on whether a task is fault-enabled?

File: ecs-agent/tmds/handlers/v4/handlers_test.go (hunk 1 above)
Comment (xxx0624) [reply ecs-agent/tmds/handlers/v4/handlers_test.go:27]: sounds good. will update this.

File: ecs-agent/tmds/handlers/v4/state/response.go (hunk 2 above)
Comment (xxx0624) [reply ecs-agent/tmds/handlers/v4/state/response.go:41]: Yes, we should. I'm thinking to add this after the HLD is finalized.

**Review by mye956 — APPROVED** [review]

File: ecs-agent/tmds/handlers/v4/handlers_test.go (hunk 1 above)
Comment (xxx0624) [reply ecs-agent/tmds/handlers/v4/handlers_test.go]: Updated.

**Review by mye956 — APPROVED** [review]

**Review by amogh09 — APPROVED** [review]

**Review by amogh09 — APPROVED** [review]

**Pull request merged**